### DISCUSSION TITLE
Relationship - sqlalchemy tries to set parents foreign key to None

### DISCUSSION THREAD

**Discussion by Rollmops** [sqlalchemy/sqlalchemy#8426] [Usage Questions] [answered]
### Describe the bug

When defining a simple One-To-Many-`relationship` (simple User-Addresses example), creating a user with two associated addresses and removing an instance from the `addresses`-relationship, sqlalchemy tries to set the foreign key of the parent (`Parent.name`) to `None`. 
I tried several `cascade`-combinations.
I red the documentations and could not find any hint explaining me that behaviour.

### To Reproduce

```python
from sqlalchemy import create_engine, Integer, Column, String
from sqlalchemy.orm import declarative_base, sessionmaker, relationship

database_engine = create_engine("sqlite:///")
database_session = sessionmaker(database_engine)()

Base = declarative_base()


class Address(Base):
    __tablename__ = "address"

    id = Column(Integer, primary_key=True)
    address = Column(String)
    user_name = Column(String(8))


class User(Base):
    __tablename__ = "user"

    name = Column(String(8), primary_key=True)

    addresses = relationship(
        Address,
        foreign_keys=[name],
        primaryjoin="Address.user_name == User.name",
        cascade="save-update, delete",
        uselist=True,
    )


Base.metadata.create_all(database_engine)

user = User(name="Hans")
address1 = Address(id=1, address="street 1", user_name="Hans")
address2 = Address(id=2, address="street 2", user_name="Hans")
user.addresses = [
    address1,
    address2,
]

database_session.add(user)
database_session.flush()

user.addresses.remove(address1)

# here is where the error occurs
database_session.flush()
```

### Error

```
AssertionError: Dependency rule tried to blank-out primary key column 'user.name' on instance '<User at 0x23ca9da47c0>'
``` 

### Versions

- OS: Windows/Linux
- Python: 3.10
- SQLAlchemy: 1.4.40
- Database: sqlite

**Comment by CaselIT** [comment, marked as answer]
Hi,

in the relationship you are setting `foreign_keys=[name],`. That's not the foreign key, you should use `Address.user_name`

**Reply by CaselIT** [reply]
It's probably easier if you set the foreign key on `user_name = Column(String(8), ForeignKey('user.name'))`.
At that point the relationship can be simply defined as `addresses = relationship(Address, cascade="save-update, delete")`
